File: packages/shared/src/types/callLog.ts
export type CallStatus = "queued" | "ringing" | "in_progress" | "ended" | "failed";

export type ResolutionStatus =
  | "resolved"
  | "follow_up_required"
  | "escalated"
  | "unknown"
  | "not_connected";

export interface ActionTaken {
  tool: string;
  ok: boolean;
  summary: string;
}

export interface CallLogSummary {
  id: number;
  astrologer_id: number;
  astrologer_name: string;
  priority: number | null;
  phone_number: string;
  triggered_by: string;
  ticket_id: number | null;
  created_ticket_id: number | null;
  status: CallStatus;
  ended_reason: string | null;
  resolution_status: ResolutionStatus | null;
  support_summary: string | null;
  created_at: string;
  ended_at: string | null;
}

export interface CallLogDetail extends CallLogSummary {
  transcript: string | null;
  suggested_solution: string | null;
  next_action: string | null;
  actions_taken: ActionTaken[] | null;
}

File: backend/app/schemas/call_log.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import CallStatus


class CallLogSummaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    astrologer_id: int
    astrologer_name: str
    priority: int | None = None
    phone_number: str
    triggered_by: str
    ticket_id: int | None = None
    created_ticket_id: int | None = None
    status: CallStatus
    ended_reason: str | None = None
    resolution_status: str | None = None
    support_summary: str | None = None
    created_at: datetime
    ended_at: datetime | None = None


class CallLogDetailRead(CallLogSummaryRead):
    transcript: str | None = None
    suggested_solution: str | None = None
    next_action: str | None = None
    actions_taken: list[dict] | None = None
